bundles/org.openhab.binding.icloud/src/main/java/org/openhab/binding/icloud/internal/ICloudConnection.java

```diff
@@ -50,7 +53,7 @@
     private final String iCloudDataRequestURL;
     private final String iCloudFindMyDeviceURL;
 
-    public ICloudConnection(String appleId, String password) throws URISyntaxException {
+    public ICloudConnection(@Nullable String appleId, @Nullable String password) throws URISyntaxException {
```

Comment:
It doesn't make sense to provide any of these as `null`, right? You should make a null-check before instantiating `ICloudConnection` (found it in `startHandler`, if any of the required credentials is null, set the thing to `OFFLINE/CONFIGURATION_ERROR` with a message stating what is wrong)

Reply:
Yes this makes sense. However if I add a null check for these parameters where this function is called, then this Nullable annotation can be removed right?
Will the tests succeeds even if there is a NonNullByDefault annotation? So these tests are non-static tests right?

Reply:
Exactly, the `@Nullable` should be removed then. If the tests fail then, there is a bug.